### ISSUE TITLE
Uncaught exception crash during search

### ISSUE THREAD

**Issue by BanditTech** [evilhero/mylar#2229]
**Describe the bug**
During automated search the system crashes and gives a traceback

**Screenshots/Logs**
`Uncaught exception: Traceback (most recent call last):
File "C:\ProgramData\mylar\mylar\logger.py", line 337, in new_run
old_run(*args, **kwargs)
File "C:\Python27\lib\threading.py", line 754, in run
self.__target(*self.__args, **self.__kwargs)
File "C:\ProgramData\mylar\mylar\helpers.py", line 3124, in search_queue
ss_queue = mylar.search.searchforissue(item['issueid'])
File "C:\ProgramData\mylar\mylar\search.py", line 1846, in searchforissue
foundNZB, prov = search_init(ComicName, IssueNumber, str(IssueYear), SeriesYear, Publisher, IssueDate, StoreDate, actissueid, AlternateSearch, UseFuzzy, ComicVersion, SARC=SARC, IssueArcID=IssueArcID, mode=mode, rsscheck=rsscheck, ComicID=ComicID, filesafe=Comicname_filesafe, allow_packs=allow_packs, oneoff=oneoff, manual=manual, torrentid_32p=TorrentID_32p, digitaldate=DigitalDate, booktype=booktype)
File "C:\ProgramData\mylar\mylar\search.py", line 360, in search_init
findit = NZB_SEARCH(ComicName, IssueNumber, ComicYear, SeriesYear, Publisher, IssueDate, StoreDate, searchprov, send_prov_count, IssDateFix, IssueID, UseFuzzy, newznab_host, ComicVersion=ComicVersion, SARC=SARC, IssueArcID=IssueArcID, RSS="no", ComicID=ComicID, issuetitle=issuetitle, unaltered_ComicName=unaltered_ComicName, allow_packs=allow_packs, oneoff=oneoff, cmloopit=cmloopit, manual=manual, torznab_host=torznab_host, torrentid_32p=torrentid_32p, digitaldate=digitaldate, booktype=booktype)
File "C:\ProgramData\mylar\mylar\search.py", line 1479, in NZB_SEARCH
searchresult = searcher(nzbprov, nzbname, mylar.COMICINFO, links, IssueID, ComicID, tmpprov, newznab=newznab_host, torznab=torznab_host, rss=RSS)
File "C:\ProgramData\mylar\mylar\search.py", line 2319, in searcher
ddl_it = ggc.parse_downloadresults(nzbid, link)
File "C:\ProgramData\mylar\mylar\getcomics.py", line 184, in parse_downloadresults
logger.fdebug('Now downloading: %s [%s] / %s ... this can take a while (go get some take-out)...' % (series, year, size))
UnboundLocalError: local variable 'year' referenced before assignment`

[mylar.zip](https://github.com/evilhero/mylar/files/2947729/mylar.zip)

**Environment (please complete the following information):**
 - OS: windows 10
 - Branch / commit: master commit from a few days back

**Additional information**
this started to happen when two import jobs got overlapped somehow, things were being downloaded automatically and I was also dropping manually downloaded collections into the folder for manual grabs. the scan folder happens to also be the same as the completed auto downloads were being placed. possibly an issue there?

**Comment by evilhero** [comment]
This has just been fixed in development today, so it's waiting for a small test base before it hits master.

That particular error is due to getcomics not returning a valid link, so you'd have to see what was downloading and manually download it outside of mylar and then manually post-process it.

Also, your log file that you're including is from the 3rd of March and is just INFO level logging which isn't helpful in diagnosing problems.

And yes keep your import directory separate from your completed directory. It will try to scan each file for a valid result and will probably end up causing problems. Also don't run more than one import process /time as there currently is no lock on those jobs to prevent duplicate run jobs.

**Comment by BanditBanditBandit** [comment]
I had debug enabled... Is the log file not being updated?

I'll try to recreate issue and post a log but it looks like you have found the problem.

On a side note, how would I install mylar so that it will update itself? I have git installed, do I just need to clone your GitHub and it will update?

**Comment by evilhero** [comment]
It looks like you're taking the log from an old location - perhaps you moved it since installation? The logs will normally default to the logs directory in the root of your mylar installation.

The most reliable way is to do a git clone and then mylar will notify you when a new update is available (it checks every 24hrs by default I believe). You just need to make sure you're running a current version of git and that your git.exe is in your Windows ENVIRONMENT Path variable.

**Comment by BanditBanditBandit** [comment]
I will do a fresh install and try to iron out those kinks. What would I do differently during git install to test the dev build?

**Comment by BanditBanditBandit** [comment]
I tried to clone the dev repo using these lines:
cd C:\Program Files\mylar
git clone --single-branch --branch development https://github.com/evilhero/mylar.git

now there is:
C:\Program Files\mylar\mylar\mylar.py

It messed with my ocd to leave it with duplicate directory! I know this is not huge deal but will it matter when updating from github that I moved it down one level? It now sits at:
C:\Program Files\mylar\mylar.py

**Comment by BanditBanditBandit** [comment]
well to keep more on the point, the development version is no longer hanging with that unknown exception. I will continue to test this version and get back to you by tomorrow.

**Comment by BanditBanditBandit** [comment]
`Uncaught exception: Traceback (most recent call last):
File "C:\Program Files\mylar\mylar\logger.py", line 337, in new_run
old_run(*args, **kwargs)
File "C:\Python27\lib\threading.py", line 754, in run
self.__target(*self.__args, **self.__kwargs)
File "C:\Program Files\mylar\mylar\helpers.py", line 3139, in search_queue
ss_queue = mylar.search.searchforissue(item['issueid'])
File "C:\Program Files\mylar\mylar\search.py", line 1855, in searchforissue
foundNZB, prov = search_init(ComicName, IssueNumber, str(IssueYear), SeriesYear, Publisher, IssueDate, StoreDate, actissueid, AlternateSearch, UseFuzzy, ComicVersion, SARC=SARC, IssueArcID=IssueArcID, mode=mode, rsscheck=rsscheck, ComicID=ComicID, filesafe=Comicname_filesafe, allow_packs=allow_packs, oneoff=oneoff, manual=manual, torrentid_32p=TorrentID_32p, digitaldate=DigitalDate, booktype=booktype)
File "C:\Program Files\mylar\mylar\search.py", line 359, in search_init
findit = NZB_SEARCH(ComicName, IssueNumber, ComicYear, SeriesYear, Publisher, IssueDate, StoreDate, searchprov, send_prov_count, IssDateFix, IssueID, UseFuzzy, newznab_host, ComicVersion=ComicVersion, SARC=SARC, IssueArcID=IssueArcID, RSS="no", ComicID=ComicID, issuetitle=issuetitle, unaltered_ComicName=unaltered_ComicName, allow_packs=allow_packs, oneoff=oneoff, cmloopit=cmloopit, manual=manual, torznab_host=torznab_host, torrentid_32p=torrentid_32p, digitaldate=digitaldate, booktype=booktype)
File "C:\Program Files\mylar\mylar\search.py", line 1487, in NZB_SEARCH
searchresult = searcher(nzbprov, nzbname, mylar.COMICINFO, links, IssueID, ComicID, tmpprov, newznab=newznab_host, torznab=torznab_host, rss=RSS)
File "C:\Program Files\mylar\mylar\search.py", line 2328, in searcher
ddl_it = ggc.parse_downloadresults(nzbid, link)
File "C:\Program Files\mylar\mylar\getcomics.py", line 279, in parse_downloadresults
vals = {'series': x['series'],
KeyError: 'series'`

[mylar.zip](https://github.com/evilhero/mylar/files/2948194/mylar.zip)
this time I have the log and had it on debug! :D

But wow it got a lot farther along in the searching before it crashed this time! fetched hundreds of new issues!

**Comment by BanditBanditBandit** [comment]
forgot to turn on debug this time, same error again using the dev build.

`

Uncaught exception: Traceback (most recent call last):File "C:\Program Files\mylar\mylar\logger.py", line 337, in new_runold_run(*args, **kwargs)File "C:\Python27\lib\threading.py", line 754, in runself.__target(*self.__args, **self.__kwargs)File "C:\Program Files\mylar\mylar\helpers.py", line 3139, in search_queuess_queue = mylar.search.searchforissue(item['issueid'])File "C:\Program Files\mylar\mylar\search.py", line 1855, in searchforissuefoundNZB, prov = search_init(ComicName, IssueNumber, str(IssueYear), SeriesYear, Publisher, IssueDate, StoreDate, actissueid, AlternateSearch, UseFuzzy, ComicVersion, SARC=SARC, IssueArcID=IssueArcID, mode=mode, rsscheck=rsscheck, ComicID=ComicID, filesafe=Comicname_filesafe, allow_packs=allow_packs, oneoff=oneoff, manual=manual, torrentid_32p=TorrentID_32p, digitaldate=DigitalDate, booktype=booktype)File "C:\Program Files\mylar\mylar\search.py", line 359, in search_initfindit = NZB_SEARCH(ComicName, IssueNumber, ComicYear, SeriesYear, Publisher, IssueDate, StoreDate, searchprov, send_prov_count, IssDateFix, IssueID, UseFuzzy, newznab_host, ComicVersion=ComicVersion, SARC=SARC, IssueArcID=IssueArcID, RSS="no", ComicID=ComicID, issuetitle=issuetitle, unaltered_ComicName=unaltered_ComicName, allow_packs=allow_packs, oneoff=oneoff, cmloopit=cmloopit, manual=manual, torznab_host=torznab_host, torrentid_32p=torrentid_32p, digitaldate=digitaldate, booktype=booktype)File "C:\Program Files\mylar\mylar\search.py", line 1487, in NZB_SEARCHsearchresult = searcher(nzbprov, nzbname, mylar.COMICINFO, links, IssueID, ComicID, tmpprov, newznab=newznab_host, torznab=torznab_host, rss=RSS)File "C:\Program Files\mylar\mylar\search.py", line 2328, in searcherddl_it = ggc.parse_downloadresults(nzbid, link)File "C:\Program Files\mylar\mylar\getcomics.py", line 279, in parse_downloadresultsvals = {'series': x['series'],KeyError: 'series'
--


`

**Comment by BanditBanditBandit** [comment]
here is another batch of logs.


[logs.zip](https://github.com/evilhero/mylar/files/2953088/logs.zip)
seems to have severaly hangups with the search fields not populating properly

**Comment by evilhero** [comment]
Yep, I'm aware of this and have it fixed currently in my local branch - just have to do more testing as there are more hangups outside of that portion of code. It has to do with the items being searched for when having the DDL provider as an option. Also, March Break kinda kills alot of free-coding time...

**Comment by BanditBanditBandit** [comment]
<3 hang in there! You are doing the nerd god's work.

**Comment by BanditBanditBandit** [comment]
wanted to also add an observation, during a scan of comics for "detective comics" it downloaded single files until it got to detective-comics-vol-3-934-958-2016-2017 which it attempted to download as a single comic after grabbing every other comic it would have gotten from this one.

**Comment by evilhero** [comment]
It actually would download that file as single comic, but it's actually a zip file (cause it's a pack). Once the pack downloads, Mylar extracts the zip file, then performs a post-processing against that entire zip extraction directory and it should post-process everything therein (as it's assigned the correct comicid prior to post-processing due to the DDL option). 

So while it says it's downloading a specific issue, it's just a logging line indicating incorrectly that it's an issue, and not a pack download (although I thought this got fixed recently). Once it snatches the pack (or queues it for download), it should mark all the issues within the pack as being Snatched within Mylar, and at that point they shouldn't attempt to be downloaded subsequently (unless of course they traverse into another pack).

**Comment by BanditBanditBandit** [comment]
i was babysitting it doing the downloading and post processing of the #'s directly before the 958 issue. Since the only way to observer the ddl backend is through the task manager XD, i was watching it download files and stop transfer for a bit and start up again. then when it got to 958 this showed up in the cache and it was downloading for ever. should be in one of those logs, sorry didnt note which one.

**Comment by evilhero** [comment]
```DDL-QUEUE : [ERROR] [Errno 13] Permission denied: u'C:\\Program Files\\mylar\\cache\\Detective Comics 934-958 (2016-2017) .zip'
10-Mar-2019 21:29:09 - INFO    :: mylar.ddl_downloader.3092 : DDL-QUEUE : [Status: False] Failed to download: {'path': None, 'success': False, 'filename': u'Detective Comics 934-958 (2016-2017) .zip'}```

Seems that you have a permissions problem in your cache folder - make sure that the user running mylar has the permission to extract (ie. execute)

**Comment by BanditBanditBandit** [comment]
it does, I dragged the finished file on top of the downloading one in the cache. the reason I was babysitting it was it seemed to be picking up all the files in the pack on its own before I got a chance to manually import them. hehe got impatient and tried to "help" it finish the dl and thats what likely gave that error. so its working, just seemed backwards for it to dl all the others then the pack.
